### PR TITLE
Update to OpenTelemetry 0.13.1

### PR DESCRIPTION
**Pull request by kenfinnigan**

### CODE REVIEW

**Comment by kenfinnigan** [comment]
Test fails locally, but not when run on own, so wanted to see what CI does

**Comment by cescoffier** [comment]
@kenfinnigan you have a CI issue:

```
2021-01-08T20:00:44.0262388Z [ERROR]   TracingPropagationTest.testFromKafkaToAppToKafka:188 
2021-01-08T20:00:44.0263583Z Expecting:
2021-01-08T20:00:44.0264107Z   <["cc94fa723755438c",
2021-01-08T20:00:44.0264665Z     "8e96f0394f410e33",
2021-01-08T20:00:44.0265085Z     "28809490762091b1",
2021-01-08T20:00:44.0265511Z     "0dcfe134d6ad3154",
2021-01-08T20:00:44.0265996Z     "232d28223056aa9a",
2021-01-08T20:00:44.0266430Z     "04c3bcb67049d641",
2021-01-08T20:00:44.0266906Z     "ec75e4cef8b3ecae",
2021-01-08T20:00:44.0268600Z     "4b6a06763026ab3e",
2021-01-08T20:00:44.0269771Z     "aee51ead0cdede1a",
2021-01-08T20:00:44.0270481Z     "7e08278a5a5a36e7"]>
2021-01-08T20:00:44.0270950Z to contain exactly (and in same order):
2021-01-08T20:00:44.0271450Z   <[]>
2021-01-08T20:00:44.0273672Z but some elements were not expected:
2021-01-08T20:00:44.0274161Z   <["cc94fa723755438c",
2021-01-08T20:00:44.0274565Z     "8e96f0394f410e33",
2021-01-08T20:00:44.0274912Z     "28809490762091b1",
2021-01-08T20:00:44.0275290Z     "0dcfe134d6ad3154",
2021-01-08T20:00:44.0275669Z     "232d28223056aa9a",
2021-01-08T20:00:44.0276049Z     "04c3bcb67049d641",
2021-01-08T20:00:44.0276454Z     "ec75e4cef8b3ecae",
2021-01-08T20:00:44.0276869Z     "4b6a06763026ab3e",
2021-01-08T20:00:44.0277281Z     "aee51ead0cdede1a",
2021-01-08T20:00:44.0277708Z     "7e08278a5a5a36e7"]>
2021-01-08T20:00:44.0277995Z 
2021-01-08T20:00:44.0278593Z [INFO] 
```

It seems that the behavior has changed.

**Review by cescoffier — CHANGES_REQUESTED** [review]
Related CI issue.

**Comment by kenfinnigan** [comment]
Yeah. I was hoping it was me locally.

What's weird is the test run on its own passes, so it's an interaction with other tests problem.

Haven't gotten to the bottom of it yet

File: pom.xml [hunk 1]
```diff
@@ -237,12 +237,12 @@
...
       <dependency>
         <groupId>io.opentelemetry</groupId>
-        <artifactId>opentelemetry-sdk</artifactId>
+        <artifactId>opentelemetry-sdk-trace</artifactId>
         <version>${opentelemetry.version}</version>
       </dependency>
       <dependency>
         <groupId>io.opentelemetry</groupId>
-        <artifactId>opentelemetry-exporters-inmemory</artifactId>
+        <artifactId>opentelemetry-sdk-testing</artifactId>
```
Comment (cescoffier) [inline pom.xml:245]: Should this dependency be in the test scope.

File: smallrye-reactive-messaging-kafka/src/main/java/io/smallrye/reactive/messaging/kafka/KafkaConnector.java [hunk 2]
```diff
@@ -98,7 +98,7 @@
 
     public static final String CONNECTOR_NAME = "smallrye-kafka";
 
-    public static final Tracer TRACER = OpenTelemetry.getTracerProvider().get("io.smallrye.reactive.messaging.kafka");
+    public static Tracer TRACER = GlobalOpenTelemetry.getTracerProvider().get("io.smallrye.reactive.messaging.kafka");
```
Comment (cescoffier) [inline smallrye-reactive-messaging-kafka/src/main/java/io/smallrye/reactive/messaging/kafka/KafkaConnector.java:101]: Why not final anymore?

File: smallrye-reactive-messaging-kafka/src/test/java/io/smallrye/reactive/messaging/kafka/tracing/TracingPropagationTest.java [hunk 3]
```diff
@@ -59,15 +59,22 @@
...
     public void setup() {
         testExporter = InMemorySpanExporter.create();
-        spanProcessor = SimpleSpanProcessor.newBuilder(testExporter).build();
-        OpenTelemetrySdk.getTracerManagement().addSpanProcessor(spanProcessor);
+        spanProcessor = SimpleSpanProcessor.builder(testExporter).setExportOnlySampled(false).build();
+        OpenTelemetrySdk.getGlobalTracerManagement().addSpanProcessor(spanProcessor);
+
+        // Without this, TRACER can be set to a tracer prior to the above configuration being active.
+        // Resulting in no traces being captured
+        KafkaConnector.TRACER = GlobalOpenTelemetry.getTracerProvider().get("io.smallrye.reactive.messaging.kafka");
```
Comment (cescoffier) [inline smallrye-reactive-messaging-kafka/src/test/java/io/smallrye/reactive/messaging/kafka/tracing/TracingPropagationTest.java:77]: Hum, I see. Not great...

**Review by cescoffier — DISMISSED** [review]

File: pom.xml (hunk 1 above)
Comment (kenfinnigan) [reply pom.xml:245]: Pretty sure I had test scope for the module importing it, but can add here for safety

File: smallrye-reactive-messaging-kafka/src/test/java/io/smallrye/reactive/messaging/kafka/tracing/TracingPropagationTest.java (hunk 3 above)
Comment (kenfinnigan) [reply smallrye-reactive-messaging-kafka/src/test/java/io/smallrye/reactive/messaging/kafka/tracing/TracingPropagationTest.java:77]: Tried a slightly different fix that retains static, currently running build now

Comment (kenfinnigan) [reply smallrye-reactive-messaging-kafka/src/test/java/io/smallrye/reactive/messaging/kafka/tracing/TracingPropagationTest.java:77]: Nothing else I try can "setup" the bits in OpenTelemetry early enough such that the static in `KafkaConnector` can "do the right thing"

**Pull request merged**